### PR TITLE
Add infra folder to minimal template

### PR DESCRIPTION
**Pull request by weikanglim**
This adds an unopinionated `main.bicep` and `main.parameters.json` when initializing from "Minimal" from `azd init`.

In azdevifying scenarios, a user could have existing files. If so, we create 'main.azd.bicep' and  'main.azd.parameters.json', logging the rename for the user to take action.

### CODE REVIEW

File: cli/azd/internal/repository/infra/main.bicep [hunk 1]
```diff
@@ -0,0 +1,25 @@
+targetScope = 'subscription'
```
Comment (ellismg) [inline cli/azd/internal/repository/infra/main.bicep:1]: Did you intend to check this file and the corresponding `main.parameters.json` in here as well?  Or should they just be in resources/minimal?

**Review by ellismg — APPROVED** [review]
LGTM - One question about duplicate files in the PR itself.

**Review by ellismg — APPROVED** [review]
LGTM - One question about duplicate files in the PR itself.

File: cli/azd/internal/repository/infra/main.bicep (hunk 1 above)
Comment (weikanglim) [reply cli/azd/internal/repository/infra/main.bicep:1]: Interesting. I may have ran init here

**Pull request merged**